pandas/core/groupby/generic.py

```diff
@@ -916,8 +916,12 @@ def transform(self, func, *args, **kwargs):
             s = klass(res, indexer)
             results.append(s)
 
-        from pandas.core.reshape.concat import concat
-        result = concat(results).sort_index()
+        # check for empty "results" to avoid concat ValueError
+        if results:
+            from pandas.core.reshape.concat import concat
+            result = concat(results).sort_index()
+        else:
+            result = Series([])
```

Comment:
Nit: For an empty `Series`, you don't need the empty list i.e. `Series()` is sufficient.